Library/Homebrew/requirement.rb

```diff
@@ -92,8 +99,15 @@ def satisfied_result_parent
     parent
   end
 
-  # Overriding {#modify_build_environment} is unsupported.
-  # Pass a block to the env DSL method instead.
+  # Pass a block to the env DSL method instead of overriding.
+  sig(:final) {
+    params(
+      env:          T.nilable(String),
+      cc:           T.nilable(String),
+      build_bottle: T::Boolean,
+      bottle_arch:  T.nilable(String),
+    ).returns(T.untyped)
```

Comment:
Note that this method currently returns `nil` or `T::Private::Types::Void::VOID`, which is clumsy/incorrect to encode in a type signature. However, we currently have a [test](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/requirement_spec.rb#L178) that relies on this being nil/non-nil depending on circumstance, so I didn't undertake a refactor. Feel free to weigh in if you want this to resolved by other means though.

Reply:
I think this is kinda weird, since `ENV#prepend_path` can technically return `nil` masked as `T::Private::Types::Void::VOID`, right?

Reply:
I'd say this should always return `nil`, or we just remove the test and make it `.void`.

Probably the latter since we don't have any test/usage that uses the return value anyways.

Reply:
Agreed that we should probably update the test here.

Reply:
I made the return type consistently void, and also updated the test to assert no error occurs (very roughly equivalent to checking for `nil` return, in this case).